azurelinuxagent/common/osutil/nsbsd.py

```diff
@@ -0,0 +1,142 @@
+#
+# Copyright 2018 Stormshield
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import azurelinuxagent.common.utils.fileutil as fileutil
+import azurelinuxagent.common.utils.shellutil as shellutil
+import azurelinuxagent.common.utils.textutil as textutil
+import azurelinuxagent.common.logger as logger
+from azurelinuxagent.common.exception import OSUtilError
+from azurelinuxagent.common.osutil.freebsd import FreeBSDOSUtil
+from azurelinuxagent.common.future import ustr
+import azurelinuxagent.common.conf as conf
+import os
+import time
+
+class NSBSDOSUtil(FreeBSDOSUtil):
+
+    resolver = None
+
+    def __init__(self):
+        super(NSBSDOSUtil, self).__init__()
+
+        if self.resolver is None:
+            # NSBSD doesn't have a system resolver, configure a python one
+
+            try:
+                import dns.resolver
+            except ImportError:
+                logger.warn("Python DNS resolver not available")
+                return
+
+            self.resolver = dns.resolver.Resolver()
+            servers = []
+            cmd = "getconf /usr/Firewall/ConfigFiles/dns Servers | tail -n +2"
+            ret, output = shellutil.run_get_output(cmd)
+            for server in output.split("\n"):
+                if server == '':
+                    break
+                server = server[:-1] # remove last '='
+                cmd = "grep '{}' /etc/hosts".format(server) + " | awk '{print $1}'"
+                ret, ip = shellutil.run_get_output(cmd)
+                servers.append(ip)
+            self.resolver.nameservers = servers
+            dns.resolver.override_system_resolver(self.resolver)
+
+    def set_hostname(self, hostname):
+        shellutil.run("/usr/Firewall/sbin/setconf /usr/Firewall/System/global SystemName {0}".format(hostname))
+        shellutil.run("/usr/Firewall/sbin/enlog")
+        shellutil.run("/usr/Firewall/sbin/enproxy -u")
+        shellutil.run("/usr/Firewall/sbin/ensl -u")
+        shellutil.run("/usr/Firewall/sbin/ennetwork -f")
+
+    def restart_ssh_service(self):
+        return shellutil.run('/usr/Firewall/sbin/enservice', chk_err=False)
+
+    def conf_sshd(self, disable_password):
+        logger.warn("SSHD already configured")
```

Comment:
… and it's not reconfigurable? There are multiple extensions that manipulate ssh settings; ideally, these changes would enable NSBSD to "play well" with them.

Reply:
ok, i'll patch that